### ISSUE TITLE
Replace extensions menu with dropdown menu

### ISSUE THREAD

**Issue by rak-phillip** [rancher/dashboard#13504]
The action menu on the extensions page needs to be updated so that it uses the new action menu component.

## Acceptance criteria

- [ ] Extensions menu on the extensions page is replaced with the ActionMenuShell component
- [ ] Keyboard navigation is supported
- [ ] Previous accessibility improvements are still present

**Comment by a-arias** [comment]
**TESTS**
Rancher : v2.11-a2ff78325d1768b0a52fa65a88c00a1f3cddace5-hea
UI: master https://github.com/rancher/dashboard/commit/16e73b16e265d49e45edd8f3186d8968bc4d9406

Fix verified, as per disscussion with Phillip on slack now user can use keyboard to navigate through the menu and using scape keys to close it ✅ 

https://github.com/user-attachments/assets/159ee0d5-d5e6-42f0-82fb-ee0cc4219be4
